File: desktop/src/renderer/api/health.ts
import { request } from "./client";

export type DependencyStatus = "ok" | "error";
export type OverallStatus = "ok" | "degraded";

export interface HealthResponse {
  status: OverallStatus;
  checkedAt: string;
  app: {
    status: "ok";
    env: string;
  };
  mysql: {
    status: DependencyStatus;
    message: string;
  };
  redis: {
    status: DependencyStatus;
    message: string;
  };
}

export function getHealth(): Promise<HealthResponse> {
  return request<HealthResponse>("/health");
}

File: backend/internal/model/health.go
package model

// HealthResponse 是健康检查接口的完整响应，包含应用和依赖状态。
type HealthResponse struct {
	Status    string           `json:"status"`
	CheckedAt string           `json:"checkedAt"`
	App       AppHealth        `json:"app"`
	MySQL     DependencyHealth `json:"mysql"`
	Redis     DependencyHealth `json:"redis"`
}

// AppHealth 描述当前应用进程自身状态。
type AppHealth struct {
	Status string `json:"status"`
	Env    string `json:"env"`
}

// DependencyHealth 描述 MySQL、Redis 等外部依赖的状态和脱敏错误信息。
type DependencyHealth struct {
	Status  string         `json:"status"`
	Message string         `json:"message"`
	Metrics map[string]any `json:"metrics,omitempty"`
}
